server/controllers/stock/core.js

```diff
@@ -773,8 +779,13 @@ async function getInventoryQuantityAndConsumption(params, monthAverageConsumptio
     const setting = await db.one(settingsql, filteredRows[0].enterprise_id);
     const nbrMonth = setting.month_average_consumption;
 
-    const startDate = new Date();
-    startDate.setDate(startDate.getDate() - 30.5 * nbrMonth);
+    let startDate = new Date();
+    let clone = moment(startDate);
+    const months = (nbrMonth - 1 > -1 && nbrMonth) ? nbrMonth - 1 : 0;
+    clone = clone.subtract(months, 'month').toDate();
```

Comment:
I'm curious why you didn't use [`moment.startOf()`](https://momentjs.com/docs/#/manipulating/start-of/) to set the month?  Was there a reason?  

This could be written: 

```js
const startDate = moment().subtract(months, 'month').startOf('month').toDate();
```